examples/complete-hosted-zone-in-diff-aws/main.tf

```diff
@@ -0,0 +1,64 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+# this is where hosted zone present/need to be created
+provider "aws" {
+  region = "us-east-1"
+  alias  = "production"
+}
+
+
+locals {
+  # Use existing (via data source) or create new zone (will fail validation, if zone is not reachable)
+  use_existing_route53_zone = true
+
+  domain = "terraform-aws-modules.modules.tf"
+
+  # Removing trailing dot from domain - just to be sure :)
+  domain_name = trimsuffix(local.domain, ".")
+}
+
+data "aws_route53_zone" "this" {
+  provider = aws.production
+
+  count = local.use_existing_route53_zone ? 1 : 0
+
+  name         = local.domain_name
+  private_zone = false
+}
+
+resource "aws_route53_zone" "this" {
+  provider = aws.production
+
+  count = ! local.use_existing_route53_zone ? 1 : 0
+  name  = local.domain_name
+}
+
+module "acm" {
+  source = "../../"
+
+  domain_name = local.domain_name
+  zone_id     = coalescelist(data.aws_route53_zone.this.*.zone_id, aws_route53_zone.this.*.zone_id)[0]
+
+  subject_alternative_names = [
+    "*.alerts.${local.domain_name}",
+    "*.something.${local.domain_name}",
+    "*.news.${local.domain_name}",
+    "*.info.${local.domain_name}",
+    "new.sub.${local.domain_name}",
+  ]
+
+  wait_for_validation = true
+
+  tags = {
+    Name = local.domain_name
+  }
+
+  providers = {
+    aws = aws
+    # use different aws provider
+    # if hosted zone is in another aws account
```

Comment:
If you're going to have a comment here, it should be something like

```
    # different provider used because hosted zone is in a different aws account
```

`use different aws provider` makes it sound like the user should use something different from `aws.route53 = aws.production`.